### PR TITLE
gnuradio3_8.pkgs.osmosdr: reduce closure size in total by ~300MB

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — APPROVED** [review]
Didn't test, but looks sensible.

**Review by bjornfor — APPROVED** [review]

File: pkgs/applications/radio/uhd/default.nix [hunk 1]
```diff
@@ -41,6 +41,8 @@ stdenv.mkDerivation rec {
   # and xxx.yyy.zzz. Hrmpf... style keeps changing
   version = "4.1.0.5";
 
+  outputs = [ "out" "dev" ];
```
Comment (doronbehar) [inline pkgs/applications/radio/uhd/default.nix:44]: This is great too, if it doesn't break any dependency.

File: pkgs/development/gnuradio-modules/osmosdr/default.nix [hunk 2]
```diff
@@ -39,6 +39,8 @@ in mkDerivation {
   inherit version src;
   disabledForGRafter = "3.9";
 
+  outputs = [ "out" "dev" ];
```
Comment (doronbehar) [inline pkgs/development/gnuradio-modules/osmosdr/default.nix:42]: With this, I'm not sure it won't break the setups of some of our `gnuradio/wrapper.nix` users..

**Review by doronbehar — COMMENTED** [review]
@SuperSandro2000 what builds did you test with this changes?

File: pkgs/development/libraries/qwt/default.nix [hunk 3]
```diff
@@ -4,6 +4,8 @@ stdenv.mkDerivation rec {
   pname = "qwt";
   version = "6.2.0";
 
+  outputs = [ "out" "dev" ];
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/qwt/default.nix:7]: If they themselves have no references to boost-dev. If they have they also need a dev output.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 what builds did you test with this changes?

`gnuradio3_8.pkgs.osmosdr`

File: pkgs/applications/radio/uhd/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/radio/uhd/default.nix:44]: Going to run a nixpkgs-review on this and if that works we are fine.,

File: pkgs/development/libraries/qwt/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/development/libraries/qwt/default.nix:7]: So if I understand correctly, with this change, referencing `qwt.dev` pulls `boost.dev` which is very large? And that's what this PR is mostly about?

[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/qwt/default.nix:7]: Yes. Similar for uhd which before also drew in boost.dev which is about 140MB added closure size.

**Comment by doronbehar** [comment]
> `Error: detected mismatched Qt dependencies:`

What package is the source of this error?

**Pull request merged**